### ISSUE TITLE
aria-roledescription is not allowed on an element without a role

### ISSUE THREAD

**Issue by jnurthen** [eBay/mindpatterns#16]
https://github.com/eBay/mindpatterns/blob/fe728715130af1de95beec57d9bcc1b681c021df/_disclosure/carousel/index.html#L73

From the ARIA spec....

"User agents MUST NOT expose the aria-roledescription property if any of the following conditions exist:

The element to which aria-roledescription is applied does not have a valid WAI-ARIA role or does not have an implicit WAI-ARIA role semantic."

**Comment by ianmcburnie** [comment]
Thanks @jnurthen . Will fix!

**Comment by ianmcburnie** [comment]
I'm wondering if a `navigation` role would be appropriate. Given that items in carousels are usally links. Or perhaps role `group`.

Also begs the question whether the word "Carousel" has any actual meaning or benefit for users...?

**Comment by jnurthen** [comment]
I don't see them as navigation links so IMO no but I guess that is situational.

**Comment by ianmcburnie** [comment]
Thanks for the clarification. I think I'm just going to remove the `aria-roledescription`, as per the first rule of ARIA :-)

**Comment by jnurthen** [comment]
to be honest I'd probably go with `role="region"` `aria-label="carousel"` . It is probably more robust that `aria-roledescription`

**Comment by jnurthen** [comment]
actually thinking more for the filmstrip one
`role="region" aria-label="related products" aria-roledescription="carousel"` is probably best

For the other one it probably depends what it is representing.

**Comment by ianmcburnie** [comment]
Thanks, my main problem with a labelled `region` is that we then get an explosion of carousels in the screen reader list of custom landmarks (eBay has a lot of carousels!). I'm trying to keep custom landmarks a little sacred ;-) but perhaps I need to rethink that. Perhaps it *would* actually be useful to have carousels showing up as custom landmarks, especially the more prominents ones such as `Your Recently Viewed Items`, `Your Watched Items`, etc. Although there is some overlap with heading structure there.

**Comment by jnurthen** [comment]
When you suggested navigation I thought that was your intent. If you don’t want it as a landmark I’d suggest ‘group’

________________________________

**Comment by ianmcburnie** [comment]
Ah yes, sorry I did confuse you (and myself) there. I'd forgotten that `navigation` would do that. I really appreciate the input! Now I have all the information, I can mull it over and make a decision. Cheers!

**Comment by ianmcburnie** [comment]
I've gone with a labelled group. Closing.
